blog/2024-11-linked-signal/README.md

```diff
@@ -0,0 +1,244 @@
+---
+title: 'Angular 19: Introducing LinkedSignal for Responsive Local State Management'
+author: Johannes Hoppe
+mail: team@angular.schule
+published: 2024-11-04
+lastModified: 2024-11-04
+keywords:
+  - Angular
+  - JavaScript
+  - Signals
+  - Reactive Programming
+  - LinkedSignal
+  - Angular 19
+  - Computed Signals 
+language: en
+thumbnail: ???.jpg
+sticky: false
+---
+
+
+In Angular 19, there's a [new experimental feature](https://github.com/angular/angular/commit/8311f00faaf282d1a5b1ddca29247a2fba94a692) called a **Linked Signal** that makes it easier to keep track of local state that depends on other signals. 
+
+It lets us create a writable signal that can automatically reset based on changes in other signals. 
+This makes it particularly useful for situations where local state needs to stay in sync with dynamic data. 
+Here’s a look at what the Linked Signal is, how it works, and some common use cases.
+
+
+## Contents
+
+* [What is LinkedSignal?](/blog/2024-11-linked-signal#what-is-linkedSignal)
+* [Basic Usage of LinkedSignal](/blog/2024-11-linked-signal#basic-usage-of-linkedSignal)
+* [Advanced Scenarios for LinkedSignal](/blog/2024-11-linked-signal#advanced-scenarios-for-linkedSignal)
+* [LinkedSignal vs. Other Signals](/blog/2024-11-linked-signal#linkedSignal-vs-other-signals)
+* [Best Practices for Using LinkedSignal](/blog/2024-11-linked-signal#best-practices-for-using-linkedSignal)
+* [Conclusion](/blog/2024-11-linked-signal#conclusion)
+
+
+<!-- > **🇩🇪 This article is available in German here: [Angular 19: Einführung von LinkedSignal für die lokale reaktionsfähige Zustandsverwaltung](https://angular-buch.com/blog/2024-11-linked-signal)** -->
+
+
+## What is LinkedSignal?
+
+A Linked Signal is an experimental feature that Angular 19 introduces to help you manage state that automatically syncs with other signals. 
+In simple terms, we receive a writable signal that resets itself when the value of its source signal changes.
+A Linked Signal can be created by using the [`linkedSignal()` factory function](https://next.angular.dev/api/core/linkedSignal).
+
+
+A Linked Signal has the following characteristics:
+
+- **Writable and Reactive**: Like a [`signal`](https://angular.dev/guide/signals#writable-signals), we can update the value of a Linked Signal manually, but it also responds to changes in its source.
+- **A Combination of Signal and Computed**: It’s like [`computed`](https://angular.dev/guide/signals#computed-signals) because it derives its value from other signals, but it stays writable, allowing us to override it when needed.
+
+
+## Basic Usage of Linked Signal
+
+To see how it works, let’s look at a complete example. 
+Here, we’re using a Linked Signal to keep track of the first book in a list, created by the `linkedSignal()` factory function.
+Whenever the list of books changes, the `firstBook` signal will automatically reset to the first book in the updated list.
```

Comment:
```suggestion
Our component has a list of books in the `books` signal.
Then we’re using a Linked Signal to keep track of the *first book* in the list, created by the `linkedSignal()` factory function.
Whenever the list of books changes, the `firstBook` signal will automatically reset to the first book in the updated list.
Up to here, all of this could have been achieved with a Computed Signal.
However, the Linked Signal makes it possible to manually set the value in the `overrideFirstBook()` method.
```